src/main/java/unicash/ui/SummaryWindow.java

```diff
@@ -69,28 +81,84 @@ public SummaryWindow() {
+                    .map(entry -> new PieChart.Data(entry.getKey(), entry.getValue()))
+                    .sorted(Comparator.comparing(PieChart.Data::getPieValue).reversed())
+                    .limit(MAX_DISPLAYED_CATEGORIES)
+                    .collect(Collectors.toCollection(FXCollections::observableArrayList));
+
+            pieChart.getData().setAll(pieChartData);
+            summaryMessageContainer.layout();
         }
+    }
 
-        pieChart.getData().clear();
+    /**
+     * Updates the line chart with expense data
+     * Note that only MAX_DISPLAYED_YEAR_MONTHS number of year-months will be displayed on the line chart
+     */
+    public void setLineGraph(HashMap<YearMonth, Double> expenseSummary) {
+        lineChart.getData().clear();
+
+        expenseSummary = fillExpenseSummary(expenseSummary);
+
+        XYChart.Series<String, Double> series = new XYChart.Series<>();
+        ObservableList<XYChart.Data<String, Double>> lineChartData = expenseSummary.entrySet().stream()
+                .sorted(Comparator.comparing(entry -> entry.getKey().toString()))
+                .filter(entry -> isValidYearMonth(entry.getKey()))
+                .map(entry -> new XYChart.Data<>(entry.getKey().format(formatter), entry.getValue()))
+                .collect(Collectors.toCollection(FXCollections::observableArrayList));
+        series.getData().addAll(lineChartData);
 
-        pieChart.getData().addAll(pieChartData);
+        lineChart.getData().add(series);
         summaryMessageContainer.layout();
+    }
+
+    /**
+     * Returns true if yearMonth is between now and EARLIEST_YEAR_MONTH, and false otherwise.
+     * Note: This function also returns true if yearMonth is the same as the year-month of now
+     * or EARLIEST_YEAR_MONTH
+     */
+    private static boolean isValidYearMonth(YearMonth yearMonth) {
+        boolean afterMostRecentYearMonth = yearMonth.compareTo(EARLIEST_YEAR_MONTH) >= 0;
+        boolean beforeCurrentYearMonth = yearMonth.compareTo(YearMonth.now()) <= 0;
+        return afterMostRecentYearMonth & beforeCurrentYearMonth;
+    }
 
+    /**
+     * Returns a defensive copy of expenseSummary with the missing year-months. A year-month is considered missing if it
+     * is after EARLIEST_YEAR_MONTH and before the current system clock's year-month.
+     */
+    private HashMap<YearMonth, Double> fillExpenseSummary(HashMap<YearMonth, Double> expenseSummary) {
+        expenseSummary = new HashMap<>(expenseSummary); // Creating a defensive copy of expenseSummary
+
+        YearMonth endYearMonth = YearMonth.now();
+        YearMonth currentYearMonth = EARLIEST_YEAR_MONTH;
+        while (currentYearMonth.isBefore(endYearMonth)) {
+            if (!expenseSummary.containsKey(currentYearMonth)) {
+                expenseSummary.put(currentYearMonth, 0.0);
+            }
+            currentYearMonth = currentYearMonth.plusMonths(1);
+        }
```

Comment:
Would it maybe be better to use a `for` loop from 0 to 12 (exclusive) to achieve this? So something like

```java
YearMonth current = YearMonth.now()
for (int i = 0; i < 12; i++) {
  YearMonth month = current.minusMonths(i);
  if (!expenseSummary.containsKey(month)) {
    expenseSummary.put(currentYearMonth, 0.0);
  }
}
```

No issues with current one though

Reply:
Agreed. I think it shortens the code and makes it a little easier to understand.